### ISSUE TITLE
[V2] topBar component buttons cause Xcode runtime crash

### ISSUE THREAD

**Issue by rgoldiez** [wix/react-native-navigation#4181]
#4105 broke components as `topBar` buttons.  I am trying to apply a component as the only item in the `leftButtons` array and the app crashes (Xcode crash).  Icon buttons work fine.

---
### Environment
* React Native Navigation version: 2.0.2583
* React Native version: 0.57.3
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): All

**Comment by guyca** [comment]
Can you add the crash log?

**Comment by yogevbd** [comment]
I think this issue fixed in this commit: https://github.com/wix/react-native-navigation/commit/30ffc07ea63f282550185403f892b114fb464d6e
can you please check on latest version? thanks

**Comment by rgoldiez** [comment]
@yogevbd - Still an issue on the latest build (2.0.2611).

**Comment by yogevbd** [comment]
@rgoldiez Can't reproduce it in the playground app, how does your options object looks like?

**Comment by rgoldiez** [comment]
See below.  These options are being applied (and causing the crash) on `setStackRoot`.  

```
{  
   component:{  
      name:'appSettings',
      passProps:{  
         selected:{  
            type:'appSettings',
            id:undefined,
            name:'App Settings'
         }
      },
      options:{  
         animations:{  
            setStackRoot:{  
               enable:false
            }
         },
         topBar:{  
            visible:true,
            animate:false,
            title:{  
               text:'App Settings'
            },
            drawBehind:false,
            leftButtons:[  
               {  
                  id:'toggleDrawer',
                  component:{  
                     name:'navIcons',
                     passProps:{  
                        name:'drawer',
                        size:24,
                        color:'#FFFFFF'
                     }
                  }
               }
            ]
         }
      }
   }
}
```

**Comment by rgoldiez** [comment]
@yogevbd - This seems to be resolved in the latest release: 2.0.2627
